sdk/servicebus/service-bus/src/receivers/receiver.ts

```diff
@@ -425,52 +402,17 @@ export class ReceiverImpl<ContextT> implements Receiver<ContextT>, SubscriptionR
    */
   async *getMessageIterator(
     options?: GetMessageIteratorOptions
-  ): AsyncIterableIterator<{
-    message: ReceivedMessage;
-    context: ContextT;
-  }> {
+  ): AsyncIterableIterator<ReceivedMessageT> {
     while (true) {
-      const { messages, context } = await this.receiveBatch(1);
+      const messages = await this.receiveBatch(1);
 
-      yield {
-        message: messages[0],
-        context
-      };
-    }
-  }
+      // TODO: punctuation?
+      if (messages == null || messages.length === 0) {
```

Comment:
The function signature on `receiveBatch()` suggests that it always returns an array, so why the null check?

Reply:
Removed, it wasn't needed. :)

Also, refactored the code out.